### PR TITLE
Miscellaneous updates

### PR DESCRIPTION
**Pull request by writinwaters**
### What problem does this PR solve?

Issue link:#1490

### Type of change

- [x] Documentation Update

### CODE REVIEW

File: docs/references/pysdk_api_reference.md [hunk 1]
```diff
@@ -1419,11 +1424,15 @@ table_obj.output(["3 * 5"]).to_result()
 ## filter
 
 ```python
-table_obj.filter(cond)
+table_obj.output(columns).filter(cond)
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1427]: remove output

Comment (JinHai-CN) [reply docs/references/pysdk_api_reference.md:1427]: table_obj => table_object

File: docs/references/pysdk_api_reference.md [hunk 2]
```diff
@@ -1419,11 +1424,15 @@ table_obj.output(["3 * 5"]).to_result()
...
 ```
 
 Creates a filtering condition expression for the current table.
 
+:::tip NOTE
+Call `filter(cond)` in a chain after calling `output(columns)` on the same table object.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1433]: Not after calling

File: docs/references/pysdk_api_reference.md [hunk 3]
```diff
@@ -1640,26 +1649,41 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [8, 10, 66]}
 ## match
 
 ```python
-table_obj.match(fields, matching_text, distance_type, topn, opt_params = None)
+table_obj.match(fields, matching_text, distance_type, options_text)
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1652]: one more options added.

Comment (JinHai-CN) [reply docs/references/pysdk_api_reference.md:1652]: "topn=5;operator=or" and "topn=5;operator=and"

File: docs/references/pysdk_api_reference.md [hunk 4]
```diff
@@ -1640,26 +1649,41 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [8, 10, 66]}
...
 ```
 
-Creates a full-text search expression.
+Performs a full-text search on the specified field(s)/column(s) and returns the most relevant rows to the provided matching text.
+
+:::danger NOTE  
+For now, it is only possible to create a full-text indext on multiple columns.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1658]: Not allow creating multiple columns index.

File: docs/references/pysdk_api_reference.md [hunk 5]
```diff
@@ -1640,26 +1649,41 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [8, 10, 66]}
...
 #### matching_text: `str`, *Required*
 
+A non-empty text string to search for. You can use various search options within the matching text, including:
+
+- Single terms: `"blooms"`
+- OR multiple terms: `"Bloom filter"`
+- Phrase search: `'"Bloom filter"'`
+- AND multiple terms: "space efficient"
+- Escaping reserved characters: "space\-efficient"
+- Sloppy phrase search: "harmful chemical"~10
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1680]: Examples will be good.

File: docs/references/pysdk_api_reference.md [hunk 6]
```diff
@@ -1640,26 +1649,41 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [8, 10, 66]}
...
+- Field-specific search: title:(quick OR brown) AND body:foobar
+
+
 #### options_text: `str`, *Required*
 
-'topn=2': Retrieve the two most relevant rows. The `topn` is `10` by default.
+Currently, only `topn` can be set in `options_text`, e.g., `"topn=10"` to retrieve the ten most relevant rows.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1686]: "topn=5;operator=or" and "topn=5;operator=and"

File: docs/references/pysdk_api_reference.md [hunk 7]
```diff
@@ -1798,44 +1823,97 @@ table_obj.fusion('match_tensor', 'topn=2', make_match_tensor_expr('t', [[0.0, -1
...
 ```
 
+Returns the current table object as a tuple.
+
+:::tip NOTE
+Call `to_result()` in a chain after (not necessarily "immediately after") `output(columns)` on the same table object.
+:::
+
+:::caution NOTE
+We recommend calling `to_df()`, `to_pl()`, or `to_arrow()` to save your results.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1840]: format

File: docs/references/pysdk_api_reference.md [hunk 8]
```diff
@@ -1798,44 +1823,97 @@ table_obj.fusion('match_tensor', 'topn=2', make_match_tensor_expr('t', [[0.0, -1
...
+
+A `pandas.DataFrame` object.
+
+## to_pl
+
 ```python
 table_obj.to_pl()
 ```
 
+Returns the current table object as a Polas DataFrame.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1876]: Similar as above

File: docs/references/pysdk_api_reference.md [hunk 9]
```diff
@@ -1798,44 +1823,97 @@ table_obj.fusion('match_tensor', 'topn=2', make_match_tensor_expr('t', [[0.0, -1
...
+
+## to_arrow
+
 ```python
 table_obj.to_arrow()
 ```
 
-After querying, these four methods above can get result into specific type. 
-`Note: output method must be executed before get result`
+Returns the current table object as an [Apache Arrow Table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table).
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1901]: Similar as above

Comment (JinHai-CN) [reply docs/references/pysdk_api_reference.md:1901]: Remove the link

File: docs/references/pysdk_api_reference.md [hunk 10]
```diff
@@ -1798,44 +1823,97 @@ table_obj.fusion('match_tensor', 'topn=2', make_match_tensor_expr('t', [[0.0, -1
...
+
+:::note
+Call `to_df()` in a chain after (not necessarily "immediately after") `output(columns)` on the same table object.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1904]: to_arrow

File: docs/getstarted/deploy_infinity_server.mdx [hunk 11]
```diff
@@ -141,13 +141,18 @@ This section provides instructions on deploying Infinity using binary package on
 Fedora/RHEL/CentOS/OpenSUSE
 ```bash
 sudo rpm -i infinity-0.3.0.dev4-x86_64.rpm
+```
+```bash
```
Comment (JinHai-CN) [inline docs/getstarted/deploy_infinity_server.mdx:145]: sudo chown -R $USER /var/infinity and increase ulimit in binary server

File: docs/references/pysdk_api_reference.md [hunk 12]
```diff
@@ -292,15 +292,15 @@ A non-empty string indicating the name of the database to retrieve.
 ### Examples
 
 ```python
-db_obj=infinity_obj.get_database("my_database")
+db_object=infinity_object.get_database("my_database")
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:295]: db_object = infinity_object

File: docs/references/pysdk_api_reference.md [hunk 13]
```diff
@@ -584,7 +580,7 @@ A non-empty string indicating the name of the index, which must adhere to the fo
   - Digits (0-9)
   - "_" (underscore)
 
-#### index_infos: `list[IndexInfo()]`, *Required*
+#### index_infos: `IndexInfo()`, *Required*
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:583]: index_infos => index_info

File: docs/references/pysdk_api_reference.md [hunk 14]
```diff
@@ -561,14 +561,10 @@ res.table_names # ['my_table, 'tensor_table', 'sparse_table']
 ## create_index
 
 ```python
-table_obj.create_index(index_name, index_infos, conflict_type = ConflictType.Error)
+table_object.create_index(index_name, index_infos, conflict_type = ConflictType.Error)
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:564]: index_infos => index_info

File: docs/references/pysdk_api_reference.md [hunk 15]
```diff
@@ -1367,63 +1347,72 @@ This method specifies the projection columns for the current table but does not
...
 ```python
 # Specify that the output should display the row number of the current table
-table_obj.output(["count(*)"]).to_result()
+table_object.output(["count(*)"]).to_result()
 ```
 
 ```python
 # Select column "num" and request all cells in this column to be displayed with their original values divided by 10
-table_obj.output(["num/10"]).to_pl()
+table_object.output(["num/10"]).to_pl()
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1387]: num / 5

File: docs/references/pysdk_api_reference.md [hunk 16]
```diff
@@ -1445,19 +1434,19 @@ This method specifies a filtering condition for the rows in the current table bu
 ### Examples
 
 ```python
-table_obj.filter("(-7 < c1 or 9 >= c1) and (c2 = 3)").to_result()
+table_object.output(["c1", "c2"]).filter("(-7 < c1 or 9 >= c1) and (c2 = 3)").to_pl()
 ```
 
 ```python
-table_obj.filter("c2 = 3").to_result()
+table_object.output(["*"]).filter("c2 = 3").to_result()
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1441]: to_pl()

File: docs/references/pysdk_api_reference.md [hunk 17]
```diff
@@ -1617,49 +1606,79 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [0.1, 0.2, 0
...
+
+- `"topn"`: `str`, *Required*  
+  Specifies the number of the most relevant rows to retrieve, e.g., `"topn=10"` to obtain the ten most relevant rows.
+- `"operator"`: `str`, *Optional*   
+  - If not specified, the search follows the default full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
+    - `&&`, `+`, `||`, `!`, `NOT`, `AND`, `OR` `-`, `,`, `~`, `^`, `:`, `""`.
+    - Escape characters like `\`, `\t`, and more.
+  - If specified, the above logical and arithmetic operators will be treated as part of the search text, and the corresponding operator will be interpolated into `matching_text`.
+    - `"operator=OR"`/`"operator=or"`: Interpolates the `OR` operator between words in `matching_text` to create a new search text.
+    - `"operator=AND"`/`"operator=and"`: Interpolates the `AND` operator between words in `matching_text` to create a new search text. Useful for searching text including code numbers like `"A01-233:BC\显卡"`, resulting in `"(A01) AND (-233) AND (:BC) AND (显卡)"`.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1677]: Remove chinese word

File: docs/references/pysdk_api_reference.md [hunk 18]
```diff
@@ -1617,49 +1606,79 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [0.1, 0.2, 0
...
 #### options_text: `str`, *Required*
 
-'topn=2': Retrieve the two most relevant rows. The `topn` is `10` by default.
+A non-empty string specifying the following search options:
+
+- `"topn"`: `str`, *Required*  
+  Specifies the number of the most relevant rows to retrieve, e.g., `"topn=10"` to obtain the ten most relevant rows.
+- `"operator"`: `str`, *Optional*   
+  - If not specified, the search follows the default full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
+    - `&&`, `+`, `||`, `!`, `NOT`, `AND`, `OR` `-`, `,`, `~`, `^`, `:`, `""`.
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1673]: `""` => `"`

File: docs/references/pysdk_api_reference.md [hunk 19]
```diff
@@ -1617,49 +1606,79 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [0.1, 0.2, 0
...
+
 #### options_text: `str`, *Required*
 
-'topn=2': Retrieve the two most relevant rows. The `topn` is `10` by default.
+A non-empty string specifying the following search options:
+
+- `"topn"`: `str`, *Required*  
+  Specifies the number of the most relevant rows to retrieve, e.g., `"topn=10"` to obtain the ten most relevant rows.
+- `"operator"`: `str`, *Optional*   
+  - If not specified, the search follows the default full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
```
Comment (JinHai-CN) [inline docs/references/pysdk_api_reference.md:1672]:
```suggestion
  - If not specified, the search follows Infinity's full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
```

File: docs/references/pysdk_api_reference.md [hunk 20]
```diff
@@ -1617,49 +1606,75 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [0.1, 0.2, 0
...
-'topn=2': Retrieve the two most relevant rows. The `topn` is `10` by default.
+A non-empty string specifying the following search options:
+
+- `"topn"`: `str`, *Required*  
+  Specifies the number of the most relevant rows to retrieve, e.g., `"topn=10"` to obtain the ten most relevant rows.
+- `"operator"`: `str`, *Optional*
+  - If not specified, the search follows Infinity's full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
+    - `&&`, `+`, `||`, `!`, `NOT`, `AND`, `OR` `-`, `(`, `)`, `~`, `^`, `:`, `"`.
+    - Escape characters like `\`, `\t`, and more.
+  - If specified, the above logical and arithmetic operators will be treated as part of the search text, and the corresponding operator will be interpolated into `matching_text`.
```
Comment (yangzq50) [inline docs/references/pysdk_api_reference.md:1671]: infinity
```suggestion
  - If specified, infinity's full-text search syntax will not be applied, and the corresponding operator will be used.
```

File: docs/references/pysdk_api_reference.md [hunk 21]
```diff
@@ -1617,49 +1606,75 @@ table_obj.match_sparse('sparse', {"indices": [0, 10, 20], "values": [0.1, 0.2, 0
...
+    - `"operator=OR"`/`"operator=or"`: Interpolates the `OR` operator between words in `matching_text` to create a new search text.
+    - `"operator=AND"`/`"operator=and"`: Interpolates the `AND` operator between words in `matching_text` to create a new search text. Useful for searching text including code numbers like `"A01-233:BC"`, resulting in `"(A01) AND (-233) AND (:BC)"`.
```
Comment (yangzq50) [inline docs/references/pysdk_api_reference.md:1673]:
```suggestion
    - `"operator=AND"`/`"operator=and"`: Interpolates the `AND` operator between words in `matching_text` to create a new search text. Useful for searching text including code numbers like `"A01-233:BC"`, resulting in `"A01" AND "-233" AND "BC"`.
```

**Review by yangzq50 — APPROVED** [review]

File: docs/references/pysdk_api_reference.md (hunk 5 above)
Comment (yuzhichang) [reply docs/references/pysdk_api_reference.md:1657]: [List of reserved characters](https://github.com/infiniflow/infinity/blob/67acc91b598626a4ba58039f78dd37c02f02e70d/src/storage/invertedindex/search/search_driver.cpp#L260):

                case ' ':
                case '+':
                case '-':
                case '=':
                case '&':
                case '|':
                case '!':
                case '(':
                case ')':
                case '{':
                case '}':
                case '[':
                case ']':
                case '^':
                case '"':
                case '~':
                case '*':
                case '?':
                case ':':
                case '\\':
                case '/':

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**